providers/targeted_refresh.md

```diff
@@ -0,0 +1,180 @@
+
+Targeted refresh requires scope to be limited, otherwise everything besides what you targeted would be deleted.  Most of this is abstracted away from the provider author but it is important to understand the fundamentals.  This is covered in way more detail in the [Provider Refresh](refresh.md) guide.
+
+### Getting Started
+
+The first thing to do is to set up the new classes that we need.  We'll need to add a `TargetCollection` class under the `Collector` and `Persister`.
+
+```ruby
+class ManageIQ::Providers::AwesomeCloud::Inventory::Collector::TargetCollection < ManageIQ::Providers::AwesomeCloud::Inventory::Collector
+end
+```
+
+```ruby
+class ManageIQ::Providers::AwesomeCloud::Inventory::Parser::TargetCollection < ManageIQ::Providers::AwesomeCloud::Inventory::Parser
+end
+```
+
+```ruby
+class ManageIQ::Providers::AwesomeCloud::Inventory::Persister::TargetCollection < ManageIQ::Providers::AwesomeCloud::Inventory::Persister
+  # Indicate that this is a targeted refresh
+  def targeted?
+    true
+  end
+
+  # This tells the Persister what to do with references, specifically this will
+  # lookup references for inventory objects which aren't in the collection which
+  # is required for targeted refresh.
+  def strategy
+    :local_db_find_missing_references
+  end
+end
+```
+
+And don't forget to add `require_nested :TargetCollection` to `app/models/manageiq/providers/awesome_cloud/inventory/collector.rb`, `app/models/manageiq/providers/awesome_cloud/inventory/parser.rb`, and `app/models/manageiq/providers/awesome_cloud/inventory/persister.rb`
+
+We also have to add a configuration setting to enable/disable targeted refresh in your provider's `config/settings.yml`:
+```
+:ems_refresh:
+  :awesome_cloud:
+    :allow_targeted_refresh: true
+```
+
+We should now have all of the scaffolding in place to start implementing the targeted collector.  The goal is that the parser and persister are basically identical in full and targeted refresh and only the collector has to change.
+
+### Targeted Collection
+
+Now that we have the basics down we can begin writing the targeted collector.  Again this is where the majority of the work will be as long as we've done everything correctly.
+
+First we have to override each collection that is defined in the base Collector:
+
+```ruby
+class ManageIQ::Providers::AwesomeCloud::Inventory::Collector::TargetCollection < ManageIQ::Providers::AwesomeCloud::Inventory::Collector
+  def images
+    []
+  end
+
+  def instances
+    []
+  end
+
+  def instance_types
+    []
+  end
+end
+```
+
+This just ensures that we won't accidentally go out and fetch the entire collection.
+
+Now we need to parse the targets that are passed in.
+
+```ruby
+class ManageIQ::Providers::AwesomeCloud::Inventory::Collector::TargetCollection < ManageIQ::Providers::AwesomeCloud::Inventory::Collector
+  def initialize(_manager, _target)
+    super
+
+    parse_targets!
+  end
+
+  ...
+
+  private
+
+  def parse_targets!
```

Comment:
NOTE: the bulk of these next three methods can probably be moved to the core collector class.  The only thing that we need from the provider is what targets are supported (e.g. MiqTemplate and Vm in this case) but that can be refactored into a method like `supported_targets`